{"timestamp":"2025-06-13T04:20:20.834015","level":"info","event":"DAG bundles loaded: dags-folder, example_dags","logger":"airflow.dag_processing.bundles.manager.DagBundlesManager"}
{"timestamp":"2025-06-13T04:20:20.836244","level":"info","event":"Filling up the DagBag from /opt/airflow/dags/etl_dag.py","logger":"airflow.models.dagbag.DagBag"}
{"timestamp":"2025-06-13T04:20:20.875701Z","level":"info","event":"Task instance is in running state","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.876093Z","level":"info","event":" Previous state of the Task instance: TaskInstanceState.QUEUED","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.876698Z","level":"info","event":"Current task name:send_email","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.876963Z","level":"info","event":"Dag name:etl_transaction_pipeline","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.877195Z","level":"info","event":"Task instance in failure state","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.877363Z","level":"info","event":"Task start","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.877514Z","level":"info","event":"Task:<Task(PythonOperator): send_email>","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.877647Z","level":"info","event":"Failure caused by Failed to send email: name 'smtplib' is not defined","chan":"stdout","logger":"task"}
{"timestamp":"2025-06-13T04:20:20.858951","level":"error","event":"Task failed with exception","logger":"task","error_detail":[{"exc_type":"Exception","exc_value":"Failed to send email: name 'smtplib' is not defined","exc_notes":[],"syntax_error":null,"is_cause":false,"frames":[{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":838,"name":"run"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1130,"name":"_execute_task"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/bases/operator.py","lineno":408,"name":"wrapper"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":212,"name":"execute"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/standard/operators/python.py","lineno":235,"name":"execute_callable"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/callback_runner.py","lineno":81,"name":"run"},{"filename":"/opt/airflow/dags/etl_dag.py","lineno":54,"name":"send_email"}]},{"exc_type":"NameError","exc_value":"name 'smtplib' is not defined","exc_notes":[],"syntax_error":null,"is_cause":false,"frames":[{"filename":"/opt/airflow/dags/etl_dag.py","lineno":48,"name":"send_email"}]}]}
{"timestamp":"2025-06-13T04:20:20.870636","level":"error","event":"Top level error","logger":"task","error_detail":[{"exc_type":"UndefinedError","exc_value":"'airflow.sdk.execution_time.task_runner.RuntimeTaskInstance object' has no attribute 'log_url'","exc_notes":[],"syntax_error":null,"is_cause":false,"frames":[{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1268,"name":"main"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1248,"name":"finalize"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/sdk/execution_time/task_runner.py","lineno":1076,"name":"_send_task_error_email"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py","lineno":473,"name":"_get_email_subject_content"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py","lineno":470,"name":"render"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/helpers.py","lineno":244,"name":"render_template_to_string"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/helpers.py","lineno":239,"name":"render_template"},{"filename":"<template>","lineno":22,"name":"root"},{"filename":"/home/airflow/.local/lib/python3.12/site-packages/jinja2/runtime.py","lineno":859,"name":"_fail_with_undefined_error"}]}]}
{"timestamp":"2025-06-13T04:20:20.886585Z","level":"warning","event":"Process exited abnormally","exit_code":1,"logger":"task"}
